### DISCUSSION TITLE
Style not applied for my Custom button

### DISCUSSION THREAD

**Discussion by samyraj42** [AvaloniaUI/Avalonia#17546] [Q&A]
### Describe the bug

I have written code for my custom button. I have created a style for that, providing a red background color.
But that style was not applied to my custom button

### To Reproduce

Step 1:

internal class CustomButton : Button
{
    public CustomButton()
    {
 
    }

   
    protected override Type StyleKeyOverride => typeof(Button);
   
}

Step2:
<Style Selector="local|CustomButton">

	<Setter Property="Background" Value="Red"/>

</Style>

Setp3:
<local:CustomButton 
					Width="150" Height="75" />

### Avalonia version

11.1.0

### OS

Windows

**Comment by timunie** [comment]
Why do you do a stylekeyoverride if you want to target your custom button? That's wrong usage here. 

In case you wanted to base on button style, you need to use ControlTheme with BasedOn set accordingly.

**Reply by samyraj42** [reply]
I will remove that one.

**Comment by maxkatz6** [comment]
Yes, by using StyleKeyOverride you told styling subsystem, that it should use "Button" as a style key.
Which is why "local|CustomButton" is not matching anymore.

**Reply by maxkatz6** [reply]
If you want to create a custom button, with some additional styled, while basing it on default Button. You should take a look at ControlTheme concept, as it allows inheriting themes via BasedOn property.

**Reply by samyraj42** [reply]
Ok, I understood. I will change it according to your advice

**Comment by samyraj42** [comment]
Thanks Guys. Now it is working fine.
